grammar/haxe.bnf

```diff
@@ -625,14 +634,8 @@ left arrayAccessExpression ::= '[' expression ']'
 
 referenceExpression ::= identifier
 {mixin="com.intellij.plugins.haxe.lang.psi.impl.HaxeReferenceImpl" implements="com.intellij.plugins.haxe.lang.psi.HaxeReference"}
-left qualifiedReferenceExpression ::= '.' referenceExpression {elementType="referenceExpression"}
-
-simpleQualifiedReferenceExpression ::= referenceExpression qualifiedReferenceExpression * { elementType="referenceExpression"}
-
-//import wildcard support
-left qualifiedReferenceExpressionOrWildcard ::= '.' referenceExpression {elementType="referenceExpression"}
-left wildcard ::= '.' '*'
-simpleQualifiedReferenceExpressionWithWildcardSupport ::= referenceExpression qualifiedReferenceExpressionOrWildcard * wildcard  { elementType="referenceExpression"}
+left qualifiedReferenceExpression ::= '.' identifier {elementType="referenceExpression" pin=2}
+private simpleQualifiedReferenceExpression ::= referenceExpression qualifiedReferenceExpression * { elementType="referenceExpression"}
```

Comment:
Ah.  You turned the reference on its head:  Now left references are contained inside of the reference, not preceding it in the PSI.  There's nothing wrong with this; we'll just have to see how things work long term.